plugins/azure/fix_plugin_azure/resource/base.py

```diff
@@ -847,6 +847,11 @@ def add_node(self, node: MicrosoftResourceType, source: Optional[Json] = None) -
             if location := self.location_lookup.get(source_location):
                 node._region = location
                 last_edge_key = self.add_edge(location, node=node)
+        elif (node_location := getattr(node, "location", None)) is not None:
```

Comment:
Isn't `if self.location` on line 843 already covering this?

Reply:
"`if self.location`" checks the `GraphBuilder` location, not the node location. In many Azure resources we have that property and can create the reference to the `AzureLocation` resource

Reply:
You are right - absolutely makes sense.